haskell/private/actions/compile.bzl

```diff
@@ -105,6 +113,30 @@ def _process_hsc_file(hs, cc, hsc_flags, hsc_inputs, hsc_file):
 
     return hs_out, idir
 
+# Not strictly for headers but that's my motivating example..
+# TODO[GL} this will fail if the header is outside of our projec root
+# what do we do then?
+def _symlink_header(hs, header, hsc_dir_raw):
+    """Symlink the given file to the output dir for hsc, preserving the "module directory structure".
+    The idea here is that hsc files often have includes with relative paths in them. We want to
+    preserve the relationship between the hsc including file and the header it includes, so in essence
+    we need to have the same directory structure.
+    Args:
+      hs: Haskell context.
+      header: The header we want to copy. Note that this is not strictly required to be a header.
+      hsc_dir_raw: The directory where we will be outputting a result with hsc2hs.
+        This is where we'll be mirroring the module structure.
+    Returns:
+      File: The generated file which is now a symlink to header.
+    """
+    # TODO[GL] fine to use rel_path_to_module here?
```

Comment:
Kind of seems like "abusing" the function. Maybe if we rename it, it'll sound better :sweat_smile: